src/gei/Commands/MigrateRepoCommand.cs

```diff
@@ -100,6 +103,18 @@ public class MigrateRepoCommand : CommandBase<MigrateRepoCommandArgs, MigrateRep
         {
             Description = "If uploading to S3, the AWS secret key. If not provided, it will be read from AWS_SECRET_KEY environment variable."
         };
+        public Option<string> AwsSessionToken { get; } = new("--aws-session-token")
+        {
+            Description = "If uploading to S3, the AWS session token."
+        };
+        public Option<string> AwsRegion { get; } = new("--aws-region")
+        {
+            Description = "If uploading to S3, the AWS resion, Default valule is us-east-1"
```

Comment:
Is the default part of the AWS SDK? I can't see it in our code - although I might be missing it.

Reply:
I wonder if it might be better to be explicit and force people to specify a region if they are using signature V4. 

S3 doesn't really have a default region per se - they are all equals - and it might lead to surprising behaviour if you rely on the default.

Reply:
> Is the default part of the AWS SDK? I can't see it in our code - although I might be missing it.
No. It's in our code in [AwsApi.cs](https://github.com/github/gh-gei/blob/main/src/Octoshift/AwsApi.cs#L14). We hardcoded it to us east 1.

> I wonder if it might be better to be explicit and force people to specify a region if they are using signature V4.
IMO defaulting to signature v4 is not a bad idea (if possible) but I still think having a default region is good, one less parameter to pass if they rally don't care about the region.

Reply:
I'd be interested to know what happens if the region you pass doesn't match the region of your bucket. 

Presumably it doesn't work, but I'm not 100% sure - I can try it tomorrow. 

If having the wrong region breaks your migration, then I think we should make it a required parameter rather than having a default which is wrong the majority of the time.

Reply:
One caveat though - if we add a required parameter, we have to think about backwards compatibility for existing users who have migration scripts and are running migrations right now. I really hope we don't need a new param!

Reply:
When using the existing version, it doesn't seem to matter if your bucket's region doesn't match the currently hardcoded region, in as much as the upload still works. The only bit that fails is subsequently retrieving the files, due to the signature issue for some regions.

It seems that the new parameters for region and signature are only there to try and solve the current signature method. If that can be solved simply by upgrading to v4 in all cases, then that would seem to do away with the need to specify the region at all.

Reply:
I just tried to set `AWSConfigsS3.UseSignatureVersion4 = true` when an `AwsApi` is instantiated, and it appears that **the region does matter** 😭 

If I set `UseSignatureVersion4 = true` and then set the region incorrectly, the generated URL has the region embedded inside it:

```
?X-Amz-Credential=AKIAYXQ3HLZ5CTZS5LEN/20230116/us-east-1/s3/aws4_request
```

...and my S3 pre-signed URLs do not work - I get an error like this:

```xml
<Error>
<Code>AuthorizationQueryParametersError</Code>
<Message>Error parsing the X-Amz-Credential parameter; the region 'us-east-1' is wrong; expecting 'eu-west-2'</Message>
<Region>eu-west-2</Region>
<RequestId>RZQ42SYGW7WS1ATW</RequestId>
<HostId>p1ICcSaLWsrFyqejE8NnFYaDfRrwVHNt8l3kLcsERJj3SGnig8QWxGoXtwgMvKpd3euL0UFh/AI=</HostId>
</Error>
```

...and before you ask, you can't just change the region in the URL because of how the signing works 😂 

I tried removing removing the `RegionEndpoint` parameter when instantiating the `AmazonS3Client` - which is supported - but I get an exception:

```
Amazon.Runtime.AmazonClientException: No RegionEndpoint or ServiceURL configured
```

In short, it seems like if we want to use V4 signing (which I think we should do in order to be compatible with a full range of regions), then we also need to accept the region as a (mandatory) input to the command.

Of course, I could be missing something, and I haven't tried the session token approach which @dylan-smith mentioned above.